src/align/LinearAlignment.cpp

```diff
@@ -223,7 +223,7 @@ std::string OptimalTranscript(const std::string& target, int j1, int j2, const s
         // Find where optimal path crosses the mid row
         //
         ublas::vector<int> sum = Sm + Sp;
-        int j = std::max_element(sum.begin() + j1, sum.begin() + j2 + 1) - sum.begin();
+        int j = std::max_element(sum.begin() + j1 - 1, sum.begin() + j2 + 1) - sum.begin();
```

Comment:
Your logic is correct.  It looks like this is the main fix---this is where things went wrong before.